pkg/core/resources.go

```diff
@@ -15,12 +15,16 @@ type (
 
 	// Resource describes a resource at the provider, infrastructure level
 	Resource interface {
-		// Provider returns name of the provider the resource is correlated to
-		Provider() string
 		// KlothoConstructRef returns AnnotationKey of the klotho resource the cloud resource is correlated to
 		KlothoConstructRef() []AnnotationKey
 		// Id returns the id of the cloud resource
-		Id() string
+		Id() ResourceId
+	}
+
+	ResourceId struct {
+		Provider string
```

Comment:
while we're here, I'd personally like to see this as a new `type ProviderId string`, and have all values pulled out to consts. There have been a couple times I've wondered "what are all the valid provider ids?", and today it's not very easy to find that set.